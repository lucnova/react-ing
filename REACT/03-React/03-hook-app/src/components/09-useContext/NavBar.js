import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
	return (
		<nav className="navbar navbar-expand-lg navbar-dark bg-dark px-3">
			<Link to="/" className="navbar-brand">
				UseContext
			</Link>

			<div className="collapse navbar-collapse">
				<ul className="navbar-nav mr-auto">
					<NavLink exact to="/" className="nav-item nav-link" activeClassName="active">
						Home
					</NavLink>
					<NavLink exact to="/about" className="nav-item nav-link" activeClassName="active">
						About
					</NavLink>
					<NavLink exact to="/login" className="nav-item nav-link" activeClassName="active">
						Login
					</NavLink>
				</ul>
			</div>
		</nav>
	);
};
