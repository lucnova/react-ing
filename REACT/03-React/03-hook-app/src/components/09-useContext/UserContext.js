// * Higher Order Component
const { createContext } = require("react");

// * Exportarlo
export const UserContext = createContext(null);