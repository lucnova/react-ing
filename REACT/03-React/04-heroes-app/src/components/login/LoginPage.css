.login-container {
	margin: 0;
	height: 100vh;
	color: #6a6f8c;
	font: 600 16px/18px 'Open Sans', sans-serif;

	background: url(https://i.ibb.co/kDs4mpq/comics-495258-1280-blur.jpg) no-repeat center center fixed;
	-webkit-background-size: cover;
	-moz-background-size: cover;
	-o-background-size: cover;
	background-size: cover;
}

.login-container .login-box {
	width: 100%;
	margin: auto;
	max-width: 525px;
	min-height: 670px;
	position: relative;
	background: rgba(0, 0, 0, 0.267);
	box-shadow: 0 12px 15px 0 rgba(0, 0, 0, 0.24), 0 17px 50px 0 rgba(0, 0, 0, 0.19);
	border-radius: 32px;
}

.login-container .login-snip {
	width: 100%;
	height: 100%;
	position: absolute;
	padding: 90px 70px 50px 70px;
	background: rgba(117, 33, 0, 0.3);
	border-radius: 32px;
}

.login-container .login-snip .login,
.login-container .login-snip .sign-up-form {
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	position: absolute;
	transform: rotateY(180deg);
	backface-visibility: hidden;
	transition: all 0.4s linear;
}

.login-container .login-snip .sign-in,
.login-container .login-snip .sign-up,
.login-container .login-space .group .check {
	display: none;
}

.login-container .login-snip .tab,
.login-container .login-space .group .label,
.login-container .login-space .group .button {
	text-transform: uppercase;
}

.login-container .login-snip .tab {
	font-size: 22px;
	margin-right: 15px;
	padding-bottom: 5px;
	margin: 0 15px 10px 0;
	display: inline-block;
	border-bottom: 2px solid transparent;
}

.login-container .login-snip .sign-in:checked + .tab,
.login-container .login-snip .sign-up:checked + .tab {
	color: #fff;
	border-color: #1161ee;
}

.login-container .login-space {
	min-height: 345px;
	position: relative;
	perspective: 1000px;
	transform-style: preserve-3d;
}

.login-container .login-space .group {
	margin-bottom: 15px;
}

.login-container .login-space .group .label,
.login-container .login-space .group .input,
.login-container .login-space .group .button {
	width: 100%;
	color: rgba(255, 255, 255, 0.966);
	display: block;
}

.login-container input {
	background-color: rgba(255, 255, 255, 0.233) !important;
}

.login-container .login-space .group .input,
.login-container .login-space .group .button {
	border: none;
	padding: 15px 20px;
	border-radius: 25px;
	background: rgba(255, 255, 255, 0.1);
}

.login-container .login-space .group input[data-type='password'] {
	text-security: circle;
	-webkit-text-security: circle;
}

.login-container .login-space .group .label {
	color: rgb(255, 255, 255);
	font-size: 12px;
}

.login-container .login-space .group .button {
	background: #ac3100;
}

.login-container .login-space .group label .icon {
	width: 15px;
	height: 15px;
	border-radius: 2px;
	position: relative;
	display: inline-block;
	background: rgba(255, 255, 255, 0.1);
}

.login-container .login-space .group label .icon:before,
.login-container .login-space .group label .icon:after {
	content: '';
	width: 10px;
	height: 2px;
	background: #fff;
	position: absolute;
	transition: all 0.2s ease-in-out 0s;
}

.login-container .login-space .group label .icon:before {
	left: 3px;
	width: 5px;
	bottom: 6px;
	transform: scale(0) rotate(0);
}

.login-container .login-space .group label .icon:after {
	top: 6px;
	right: 0;
	transform: scale(0) rotate(0);
}

.login-container .login-space .group .check:checked + label {
	color: #fff;
}

.login-container .login-space .group .check:checked + label .icon {
	background: #1161ee;
}

.login-container .login-space .group .check:checked + label .icon:before {
	transform: scale(1) rotate(45deg);
}

.login-container .login-space .group .check:checked + label .icon:after {
	transform: scale(1) rotate(-45deg);
}

.login-container .login-snip .sign-in:checked + .tab + .sign-up + .tab + .login-space .login {
	transform: rotate(0);
}

.login-container .login-snip .sign-up:checked + .tab + .login-space .sign-up-form {
	transform: rotate(0);
}

.login-container *,
.login-container :after,
.login-container :before {
	box-sizing: border-box;
}

.login-container .clearfix:after,
.login-container .clearfix:before {
	content: '';
	display: table;
}

.login-container .clearfix:after {
	clear: both;
	display: block;
}

.login-container a {
	color: inherit;
	text-decoration: none;
}

.login-container .hr {
	height: 2px;
	margin: 60px 0 50px 0;
	background: rgba(255, 255, 255, 0.2);
}

.login-container .foot {
	text-align: center;
	color: white;
}

.login-container .card {
	min-width: 256px;
	background-color: rgba(0, 0, 0, 0);
	border: 0px;
}

.login-container ::placeholder {
	color: #b3b3b3;
}
