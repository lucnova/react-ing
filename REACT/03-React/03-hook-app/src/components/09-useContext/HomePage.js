import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomePage = () => {
	const userContext = useContext(UserContext);
	const { user } = userContext;

	return (
		<>
			<h1>Home Page</h1>
			<hr />

			<pre>{JSON.stringify(user, null, 4)}</pre>
		</>
	);
};
