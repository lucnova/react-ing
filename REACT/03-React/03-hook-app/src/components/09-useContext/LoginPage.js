import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginPage = () => {
	const { setUser } = useContext(UserContext);

	return (
		<>
			<h1>Login Page</h1>
			<hr />

			<button
				className="btn btn-primary"
				onClick={() =>
					setUser({
						id: '123ABC',
						username: 'lucnova',
						email: 'alanugb@gmail.com',
					})
				}
			>
				Login
			</button>

			<button className="btn btn-warning" onClick={() => setUser({})}>
				Logout
			</button>
		</>
	);
};
