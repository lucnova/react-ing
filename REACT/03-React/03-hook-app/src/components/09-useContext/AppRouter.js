import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
	// * El Switch devuelve lo primero que se indique; si pones Route path='/', no importa si te dirijes a /about;
	//       coinciden el / y es suficiente para el Switch,
	//      Para hacerlo exacto, colocar 'exact' como propiedad.
	return (
		<>
			<Router>
				<div>
					<NavBar />
					<Switch>
						<Route exact path="/home" component={HomePage} />
						<Route exact path="/login" component={LoginPage} />
						<Route exact path="/about" component={AboutPage} />
						<Route exact path="/" component={HomePage} />
					</Switch>
				</div>
			</Router>
		</>
	);
};
