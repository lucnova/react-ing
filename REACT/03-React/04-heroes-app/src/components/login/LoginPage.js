import React from 'react';
import './LoginPage.css';

export const LoginPage = () => {
	const handleLogin = () => {
		console.log('yo');
	};

	return (
		<div className="container">
			<div className="row">
				<div className="col-md-6 mx-auto p-0">
					<div className="card rounded">
						<div className="login-box">
							<div className="login-snip">
								<input id="tab-1" type="radio" name="tab" className="sign-in" defaultChecked />
								<label htmlFor="tab-1" className="tab">
									Iniciar Sesión
								</label>
								<input id="tab-2" type="radio" name="tab" className="sign-up" hidden={true} />
								<label htmlFor="tab-2" className="tab" hidden={true}>
									Registrarse
								</label>
								<div className="login-space">
									<div className="login">
										<div className="group">
											<label htmlFor="user" className="label">
												Usuario
											</label>
											<input name="user" type="text" className="input" placeholder="Ingresa tu Usuario" />
										</div>
										<div className="group">
											<label htmlFor="pass" className="label">
												Contraseña
											</label>
											<input name="pass" type="password" className="input" data-type="password" placeholder="*********" />
										</div>
										<div className="group">
											<button className="button" onClick={handleLogin}>
												Ingresar
											</button>
										</div>
										<div className="hr" />
									</div>
									<div className="sign-up-form"></div>
								</div>
							</div>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};
