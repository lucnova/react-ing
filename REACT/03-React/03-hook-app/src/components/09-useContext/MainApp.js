import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

// * Establecer el contexto a todo el componente principal.

export const MainApp = () => {
	const user = {
		id: '12ABC',
		username: 'lucnova',
		email: 'nlnovae@gmail.com',
	};

	return (
		<UserContext.Provider value={user}>
			<AppRouter />
		</UserContext.Provider>
	);
};
