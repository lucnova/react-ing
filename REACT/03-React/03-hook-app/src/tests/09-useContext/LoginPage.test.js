import { mount, shallow } from 'enzyme';
import { LoginPage } from '../../components/09-useContext/LoginPage';
import { UserContext } from '../../components/09-useContext/UserContext';

describe('Testing on LoginPage', () => {
	// * Siempre al hacer pruebas con funciones, jest.fn() debe usarse para definirse el wrapper
	const setUser = jest.fn();

	const wrapper = mount(
		<UserContext.Provider
			value={{
				setUser,
			}}
		>
			<LoginPage />
		</UserContext.Provider>
	);

	test('should display correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should execute setUser as expected', () => {
		wrapper.find('button').at(0).prop('onClick')();
        // * Nota, daba erro r porque tengo dos botones; eligo con el 'at()'
		//wrapper.find('button').prop('onClick')();
        //wrapper.find(LoginPage).find('button').simulate('click');

		expect(setUser).toBeCalledWith({
			id: '123ABC',
			username: 'lucnova',
			email: 'alanugb@gmail.com',
		});
	});
});
