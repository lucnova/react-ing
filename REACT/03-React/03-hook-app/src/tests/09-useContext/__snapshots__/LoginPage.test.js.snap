// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on LoginPage should display correctly 1`] = `
Array [
  <h1>
    Login Page
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Logout
  </button>,
]
`;
